CodeLab.cs

```diff
@@ -47,13 +47,50 @@ public class PluginSupportInfo : IPluginSupportInfo
     [PluginSupportInfo(typeof(PluginSupportInfo), DisplayName = "CodeLab")]
     public class CodeLab : Effect
     {
+        [DllImport("gdi32.dll")]
+        static extern int GetDeviceCaps(IntPtr hdc, int nIndex);
+        public enum DeviceCap
+        {
+            VERTRES = 10,
+            LOGPIXELSY = 90,
+            DESKTOPVERTRES = 117,
+        }
+
+        private static float getScalingFactor()
+        {
+            int LogicalScreenHeight;
+            int PhysicalScreenHeight;
+            int logpixelsy;
+            using (Graphics g = Graphics.FromHwnd(IntPtr.Zero))
+            {
+                IntPtr desktop = g.GetHdc();
+                LogicalScreenHeight = GetDeviceCaps(desktop, (int)DeviceCap.VERTRES);
+                PhysicalScreenHeight = GetDeviceCaps(desktop, (int)DeviceCap.DESKTOPVERTRES);
+                logpixelsy = GetDeviceCaps(desktop, (int)DeviceCap.LOGPIXELSY);
+                g.ReleaseHdc();
+            }
+            float screenScalingFactor = (float)PhysicalScreenHeight / (float)LogicalScreenHeight;
+            float dpiScalingFactor = (float)logpixelsy / (float)96;
+            return Math.Max(screenScalingFactor, dpiScalingFactor);
+        }
+
```

Comment:
Let's just wait until Rick makes that `UIScaleFactor` utility class available. Then we can just use the static `UIScaleFactor.Current` property.